packages/clerk-js/src/core/clerk.ts

```diff
@@ -708,8 +708,16 @@ export class Clerk implements ClerkInterface {
     // to keep the organization id that the session had.
     const shouldSwitchOrganization = organization !== undefined;
     if (newSession && shouldSwitchOrganization) {
-      const organizationId = typeof organization === 'string' ? organization : organization?.id;
-      newSession.lastActiveOrganizationId = organizationId || null;
+      const organizationIdOrSlug = typeof organization === 'string' ? organization : organization?.id;
+
+      if (organizationIdOrSlug?.startsWith('org_')) {
```

Comment:
If the string value starts with `org_`, then it's automatically an `organization_id` and we don't have to do extra checks.